frontend/src/features/admin-form/responses/ResponsesPage/storage/useDecryptionWorkers.ts

```diff
@@ -191,95 +195,99 @@ const useDecryptionWorkers = ({
                 }
                 return decryptResult
               })
+              // Step 3: Update the decryption count
               .then((decryptResult) => {
                 decryptionProgress += 1
-                onProgress(decryptionProgress)
                 return decryptResult
+              })
+              // Step 4: Save the downloaded and decrypted attachment blobs for each submission (if required).
+              // This step is done with delays in between groups of files to space out downloads to avoid browser blocking downloads.
+              .then(async (decryptResult) => {
+                if (downloadAttachments && decryptResult.downloadBlob) {
+                  // Ensure attachments downloads are spaced out to avoid browser blocking downloads
+                  if (
+                    decryptionProgress % ATTACHMENT_DOWNLOAD_CONVOY_SIZE ===
+                    0
+                  ) {
+                    const now = new Date().getTime()
+                    const elapsedSinceXDownloads =
+                      now - timeSinceLastXAttachmentDownload
+
+                    const waitTime = Math.max(
+                      0,
+                      ATTACHMENT_DOWNLOAD_CONVOY_MINIMUM_SEPARATION_TIME -
+                        elapsedSinceXDownloads,
+                    )
+                    if (waitTime > 0) {
+                      await waitForMs(waitTime)
+                    }
+                    timeSinceLastXAttachmentDownload = now
+                  }
+                  await downloadResponseAttachment(
+                    decryptResult.downloadBlob,
+                    decryptResult.id,
+                  )
+                }
+                return decryptResult
+              })
+              .catch((err) => {
```

Comment:
note: if the download has failed - err thrown here, it is still possible for the err count to not reflect the failure, since the error count is based on the above csvRecord